SafeguardSessions/SafeguardSessions.pq

```diff
@@ -55,35 +58,60 @@ shared FetchSessions = (
             from = from,
             to = to
         ],
-        data = GetData(sps_ip, credentials, query, mocks)
+        data = GetData(spsIp, credentials, query, customFirmware ?? false, mocks)
     in
         data;
 
-shared GetData = (sps_ip as text, credentials as record, query as record, optional mocks as nullable record) =>
+shared GetData = (
+    spsIp as text, credentials as record, query as record, customFirmware as logical, optional mocks as nullable record
+) =>
     let
         startTime = DateTimeZone.LocalNow(),
-        spsUrl = UrlBuilder.GenerateBaseUrl("https", sps_ip),
+        spsUrl = UrlBuilder.GenerateBaseUrl("https", spsIp),
         authResponse =
             if IsMocked(mocks, "Authenticate") then
                 mocks[Authenticate][Value]
             else
                 Authenticate(spsUrl, credentials, mocks),
+        sessionId = ValidateSPSVersion(
+            spsUrl,
+            BaseAuthentication.MineSessionIdFromCookie(Request.GetHeader(Value.Metadata(authResponse), "Set-Cookie")),
+            customFirmware,
+            mocks
+        ),
         response =
             if IsMocked(mocks, "FetchData") then
                 mocks[FetchData][Value]
             else
-                FetchData(
-                    spsUrl,
-                    UrlBuilder.GetDefaultQuery() & query,
-                    BaseAuthentication.MineSessionIdFromCookie(
-                        Request.GetHeader(Value.Metadata(authResponse), "Set-Cookie")
-                    ),
-                    mocks
-                ),
+                FetchData(spsUrl, UrlBuilder.GetDefaultQuery() & query, sessionId, mocks),
         data = HandleIfThereIsAnError(try (response)),
         result = BuildSPSResult(data, startTime)
     in
         result;
+
+ValidateSPSVersion = (
```

Comment:
Working on it!

Reply:
I found a small issues. We only need the version at the last step, where we build the result table.

If version is only called at that step, then all the intermediate queries will be executed. I'll try putting the version in the meta.

Reply:
Yeah I forgot about the lazy evaluation. The it doesn't worth the effort, since the Info table contains the sps IP, so based on that, the user can find out after logging in to the appliance.

Reply:
Almost done. The README.md still needs to be updated as when we write about the Info table we mention that it only contains information related to the data fetching process:
https://github.com/OneIdentity/SafeguardPowerBI/blob/main/SafeguardSessions/README.md?plain=1#L18

Reply:
Extended.